examples/hubert/dataset/hubert_dataset.py

```diff
@@ -309,14 +310,20 @@ def _collate_audio_label(
         kernel_size = 25
         stride = 20
         sample_rate = 16  # 16 per millisecond
-        if waveform.shape[1] > audio_size:
-            diff = waveform.size(1) - audio_size
+        audio_start = 0
+        waveform = waveform[0]
+        if waveform.shape[0] > audio_size:
+            diff = waveform.size(0) - audio_size
             audio_start = torch.randint(diff, size=(1,)) if rand_crop else 0
-            label_start = torch.div(
-                audio_start - kernel_size * sample_rate, stride * sample_rate, rounding_mode="floor"
-            )
-            label_size = torch.div(audio_size - kernel_size * sample_rate, stride * sample_rate, rounding_mode="floor")
-            waveform = waveform[:, audio_start : audio_start + audio_size]
-            label = label[label_start : label_start + label_size]
-            length = audio_size
+        else:
+            audio_size = waveform.shape[0]
+        label_start = torch.max(
+            torch.div(audio_start - kernel_size * sample_rate, stride * sample_rate, rounding_mode="floor") + 1,
+            torch.tensor([0]),
+        )
+        label_size = torch.div(audio_size - kernel_size * sample_rate, stride * sample_rate, rounding_mode="floor") + 1
```

Comment:
is this correct when `audio_size - kernel_size * sample_rate` is a multiple of `stride * sample_rate`?

Reply:
Yes, it's correct as there is padding in the convolutional layer of wav2vec2 and hubert models. The way to compute the label_size is from the [length computation in wav2vec2](https://github.com/pytorch/audio/blob/main/torchaudio/models/wav2vec2/components.py#L64).

Reply:
Can you add unit test?

Reply:
sure

Reply:
> is this correct when `audio_size - kernel_size * sample_rate` is a multiple of `stride * sample_rate`?

@nateanl `audio_size` is not precise naming. (it could mean size in bytes from other software engineers.) Can you use `frames` to refer to the number of frames? In torchaudio, the standard is to use `frame` when referring to the length of audio in channel-agnostic way, and `sample` to refer to the number of samples across channels. (i.e. `num_frames` * `num_channels` == `num_samples`)